In [12]:
import os
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

star_df = pd.read_csv('https://drive.google.com/uc?id=1BQVc6MHjQFtDC9iP1isT_K4ojVe_Oil-')

star_df.sample(10)

Temperature (K)  Luminosity(L/Lo)  Radius(R/Ro)  Absolute magnitude(Mv)  \
132             3100           0.00800        0.3100                   11.17   
87             10980           0.00074        0.0087                   11.19   
204            18340           0.00134        0.0124                   11.22   
231            38234      272830.00000     1356.0000                   -9.29   
70              3345           0.02100        0.2730                   12.30   
71              3607           0.02200        0.3800                   10.12   
47              3574      200000.00000       89.0000                   -5.24   
23              8500           0.00050        0.0100                   14.50   
170             3570      320000.00000     1480.0000                   -7.58   
127             2861           0.00019        0.0899                   16.71   

     Star type  Star color Spectral Class  
132          1         Red              M  
87           2  Blue-White              B  
204          2        Blue              B  
231          5        Blue              O  
70           1         Red              M  
71           1         Red              M  
47           4         Red              M  
23           2       White              A  
170          5         Red              M  
127          0         Red              M

In [13]:
star_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Temperature (K)         240 non-null    int64  
 1   Luminosity(L/Lo)        240 non-null    float64
 2   Radius(R/Ro)            240 non-null    float64
 3   Absolute magnitude(Mv)  240 non-null    float64
 4   Star type               240 non-null    int64  
 5   Star color              240 non-null    object 
 6   Spectral Class          240 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 13.3+ KB


Observations:
Total Rows = 240
Feature columns = 6
Target Columns = 1

Absolute Temperature(K)
Relative Luminosity(L/Lo)
Relative Radius(R/Ro)
Absolute MAgnitude(Mv)
Star Color
Spectral Class
Star Type(Target)

Here , Lo = 3.828 x 10^26 Watts (Average Luminoisity of Sun)
       Ro = 6.9551 x 10^8 m (Average Radius of Sun)
       
Categorical Features: Star Color and Spectral Class



